src/StaticWebAssetsSdk/Tasks/Data/StaticWebAssetPathPattern.cs

```diff
@@ -190,6 +190,8 @@ public static StaticWebAssetPathPattern Parse(string rawPath, string assetIdenti
                         }
                         else if (!string.IsNullOrEmpty(part.Value))
                         {
+                            // Token was embedded, so add it to the dictionary.
+                            dictionary[part.Name] = part.Value;
```

Comment:
When the token was embedded in the expression (for example, `file#[.{fingerprint=value}].ext` we weren't adding it to the tokens returned, so it wasn't being added as a property and we weren't setting the caching headers correctly because of that.